src/board/system76/common/peci.c

```diff
@@ -56,6 +56,22 @@ static struct Fan __code FAN = {
     .interpolate = SMOOTH_FANS != 0,
 };
 
+int16_t peci_set_fan_curve(uint8_t count, struct FanPoint * points) {
+    if (count != FAN.points_size) {
+        TRACE("PECI: Incorrect number of fan points: %d, expected %d\n",
+            count, FAN.points_size);
+        return -1;
+    }
+
+    // TODO: check for curve validity? Or assume BIOS / user are sane?
```

Comment:
Yes please do the check

Reply:
Added a check for curve sanity in smfi.c - comment removed